include/vara/Feature/Feature.h

```diff
@@ -52,7 +63,11 @@ class Feature {
 
   [[nodiscard]] bool isOptional() const { return Opt; }
 
-  [[nodiscard]] bool isRoot() const { return Parents.empty(); }
+  [[nodiscard]] virtual bool isBinary() const = 0;
+
+  [[nodiscard]] virtual bool isNumeric() const = 0;
```

Comment:
Why to we implement specific check functions when we use inheritance? If we need this information we should impl llvm style type info with kind enums. Then we can write `llvm::isa<BinaryFeature *>()`